apps/concierge_site/lib/controllers/unsubscribe_controller.ex

```diff
@@ -0,0 +1,19 @@
+defmodule ConciergeSite.UnsubscribeController do
+  use ConciergeSite.Web, :controller
+  alias AlertProcessor.Model.Trip
+  @thirty_days_in_seconds 2_592_000
+
+  def update(conn, %{"id" => encrypted_user_id}) do
+    # We're not using a separate salt here because we're not
+    # encrypting passwords and don't need the extra security
+    {:ok, user_id} =
+      Plug.Crypto.decrypt(conn.secret_key_base, conn.secret_key_base, encrypted_user_id,
+        max_age: @thirty_days_in_seconds
+      )
+
+    Trip.get_trips_by_user(user_id)
+    |> Enum.each(&Trip.pause(&1, user_id))
```

Comment:
suggestion: As a "belt and suspenders" approach, I wonder if it makes sense to also set the user's `communication_mode` to `none`.

Reply:
That's a good idea, I'll put that in.